tests/sensors/test_time_sensor.py

```diff
@@ -63,7 +63,6 @@ def test_task_is_deferred(self):
 
         assert isinstance(exc_info.value.trigger, DateTimeTrigger)
         assert exc_info.value.trigger.moment == timezone.datetime(2020, 7, 7, 10)
-        assert exc_info.value.method_name == "execute_complete"
```

Comment:
Should we still check the method name is `__trigger_exit__` here?

Reply:
In this case the method_name would still be `execute_complete` but it would not run when `end_from_trigger` is True.

Reply:
If that's the case, we probably shouldn't remove this line. It's still the correct value, but it just does not run. Not sure whether we have a way to check it's not running here. I guess not. probably another test case

Reply:
The current implementation has `method_name` set as `execute_complete` for exiting a task from the worker option. Otherwise, if exiting a task from the trigger, the `method_name` doesn't matter. Adding an additional test to check for the `method_name` when `end_from_trigger=True` would require changing the `method_name` in the sensor as well.

Reply:
I'm not sure I get `Adding an additional test to check for the method_name when end_from_trigger=True would require changing the method_name in the sensor as well`. I thought it would still just be `execute_complete` but something not used.

Reply:
I mean if you look at https://github.com/apache/airflow/blob/4bba0d1f4a6ff7f34c786e4a56ad1934cb923e66/airflow/sensors/time_sensor.py#L78-L81 we are always passing `execute_complete` as `method_name` regardless of `end_from_trigger=True` or `end_from_trigger=False`. We implemented this to facilitate the user's ability to end the task by the worker for this sensor. That is not required for `end_from_trigger=True`.

Reply:
yep, so I think we could still check `assert exc_info.value.method_name == "execute_complete"`. Don't need to change the `method_name`. `end_from_trigger` or not, it's `execute_complete`

Reply:
Sure. I will add `assert exc_info.value.method_name == "execute_complete"`

Reply:
changed it: https://github.com/apache/airflow/pull/40084/commits/a7a2c93ea1e7d73315363c6f571cadb3a93c0963

Reply:
Thanks for taking time 🙏